src/dialects/oracle/query-generator.js

```diff
@@ -0,0 +1,1443 @@
+      ' END IF;',
+      'END;'
+    ]);
+  }
+
+  /*
+    Modifying the indexname so that it is prefixed with the schema name
+    otherwise Oracle tries to add the index to the USER schema
+   @overide
+  */
+  addIndexQuery(tableName, attributes, options, rawTablename) {
+    if (typeof tableName !== 'string' && attributes.name) {
+      attributes.name = `${tableName.schema}.${attributes.name}`;
+    }
+    return super.addIndexQuery(tableName, attributes, options, rawTablename);
+  }
+
+  addConstraintQuery(tableName, options) {
+    options = options || {};
+
+    const constraintSnippet = this.getConstraintSnippet(tableName, options);
+
+    if (typeof tableName === 'string') {
+      tableName = this.quoteIdentifier(tableName);
+    } else {
+      tableName = this.quoteTable(tableName);
+    }
+
+    return `ALTER TABLE ${tableName} ADD ${constraintSnippet};`;
+  }
+
+  addColumnQuery(table, key, dataType) {
+    dataType.field = key;
+
+    const attribute = Utils.joinSQLFragments([
+      this.quoteIdentifier(key),
+      this.attributeToSQL(dataType, {
+        context: 'addColumn'
+      })
+        .replace('ATTRIBUTENAME', this.quoteIdentifier(key))
+        .replace(/'/g, "'")
+    ]);
+
+    return Utils.joinSQLFragments([
+      'ALTER TABLE',
+      this.quoteTable(table),
+      'ADD',
+      attribute
+    ]);
+  }
+
+  removeColumnQuery(tableName, attributeName) {
+    return Utils.joinSQLFragments([
+      'ALTER TABLE',
+      this.quoteTable(tableName),
+      'DROP COLUMN',
+      this.quoteIdentifier(attributeName),
+      ';'
+    ]);
+  }
+
+  /**
+   * Function to add new foreign key to the attribute 
+   * Block for add and drop foreign key constraint query
+   * taking the assumption that there is a single column foreign key reference always
+   * i.e. we always do - FOREIGN KEY (a) reference B(a) during createTable queryGenerator
+   * so there would be one and only one match for a constraint name for each column
+   * and every foreign keyed column would have a different constraint name
+   * Since sequelize doesn't support multiple column foreign key, added complexity to
+   * add the feature isn't needed
+   *
+   * @param {Array} sql Array with sql blocks
+   * @param {string} definition The operation that needs to be performed on the attribute
+   * @param {string|object} table The table that needs to be altered
+   * @param {string} attributeName The name of the attribute which would get altered
+   */
+  _alterForeignKeyConstraint(sql, definition, table, attributeName) {
+    const [tableName, schemaName] = this.getSchemaNameAndTableName(table);
+    const attributeNameConstant = wrapSingleQuote(this.getCatalogName(attributeName));
+    const schemaNameConstant = table.schema ? wrapSingleQuote(this.getCatalogName(schemaName)) : 'USER';
+    const tableNameConstant = wrapSingleQuote(this.getCatalogName(tableName));
+    const getConsNameQuery = [
+      'select constraint_name into cons_name',
```

Comment:
May as well be consistent with the case of keywords.  In most places the dialect uppercases them.

Reply:
Done